### PR TITLE
Server: Use multi-task for embeddings endpoint

### PR DESCRIPTION
**Pull request by ngxson**
As discussed in https://github.com/ggerganov/llama.cpp/pull/5939#pullrequestreview-1924944420 , using multi-task for multi prompts in embeddings endpoint is preferable.

I also make the code less verbose for this function. There're maybe other ways to make the code even shorter, but I prefer stay this way to keep it readable (also easy to understand what the code does). But suggestions are welcome for this.

The next task for me will be to see if we can reuse part of code inside `handle_completions` for `handle_chat_completions` to prevent code duplication.

### CODE REVIEW

**Comment by ngxson** [comment]
@ggerganov That's weird, the command above works on my computer:

```
make clean && make server && ./server -m ../bert-bge-small.gguf --embedding --port 6900 -v --log-format text -c 4096 -b 4096
```

```
curl -sS --data '{"content":"1234567890"}' http://127.0.0.1:6900/embedding
{"embedding":[-0.03480084612965584,-0.07584927976131439,-0.006564121227711439,-0.017552589997649193,-0.05999302491545677,-0.0035985081922262907,0.029658228158950806,0.019941214472055435,-0.0001966610725503415,-0.0010271853534504771,0.0118120647............
```

Link to model: https://huggingface.co/ggml-org/models/blob/main/bert-bge-small/ggml-model-f16.gguf

Do you have the server log?

**Comment by ngxson** [comment]
That turns out to be true, the [windows test](https://github.com/ggerganov/llama.cpp/actions/runs/8239810033/job/22533870765) passed. Sometimes compilers do weird things...

File: examples/server/server.cpp
Comment (phymbert) [inline examples/server/server.cpp:3395]: `n_predict` is useless here

Comment (ngxson) [reply examples/server/server.cpp:3395]: I'm not sure what's the purpose of `{"n_predict", 0}` here so I just leave it as-is. (It was there from a long time ago)

I think `n_predict` must be explicitly set to 0 because we're not generating new tokens on embedding mode, only evaluating them. Maybe it's not the best place to set `n_predict` here, should be in `request_completion`.

But this part I'm not 100% sure so I'll wait for confirmation from @ggerganov

**Review by ggerganov — APPROVED** [review]

File: examples/server/server.cpp
Comment (ggerganov) [reply examples/server/server.cpp:3395]: Yes, seems useless - if a problem arises from removing it we should fix embedding processing to not depend in any way on `n_predict`

Comment (ngxson) [reply examples/server/server.cpp]: Yeah seems like it doesn't break anything. I pushed a new commit to remove this. Will merge when the CI test passed.

**Review by phymbert — APPROVED** [review]

**Comment by ngxson** [comment]
@phymbert I noticed that embedding test with `bert-bge-small` is quite slow particularly with `LLAMA_SANITIZE_THREAD` (usually takes 30 minutes to finish). I'm wondering if we can skip embedding test on `LLAMA_SANITIZE_THREAD`,while keeping it on `LLAMA_SANITIZE_ADDRESS` and `LLAMA_SANITIZE_UNDEFINED`

**Comment by phymbert** [comment]
Yes I am on it already, it will run on master scheduled only. @ggerganov ok to remove it for PR ?

**Comment by ggerganov** [comment]
Yes

**Pull request merged**
